patches/tModLoader/Terraria.ModLoader.IO/TagSerializable.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Reflection;
+
+namespace Terraria.ModLoader.IO
+{
+	public interface TagSerializable
+	{
+		TagCompound SerializeData();
+	}
+
+	public interface ITagDeserializer<out T> where T : TagSerializable
```

Comment:
No-one's going to override this, just use TagDeserializer, so why don't we just use Func<TagCompound, T> for the DESERIALIZER field